File: LspServer/Handlers/AssetProperties/AssetProperty.cs
using Newtonsoft.Json;
using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;

namespace DanielWillett.UnturnedDataFileLspServer.Handlers.AssetProperties;

public class AssetProperty
{
    [JsonProperty("key")]
    public required string Key { get; set; }

    [JsonProperty("range"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public Range? Range { get; set; }

    [JsonProperty("value")]
    public object? Value { get; set; }

    [JsonProperty("description")]
    public string? Description { get; set; }

    [JsonProperty("markdown")]
    public string? Markdown { get; set; }
}

File: unturned-dat/src/spec/asset-property.ts
import { Range } from 'vscode';

export interface AssetProperty
{
    key: string,
    range: Range | undefined,
    value: any,
    description: string | null | undefined,
    markdown: string | null | undefined;
    ordinal: number | undefined;
    children: AssetProperty[] | undefined;
}